library/channel.c

```diff
@@ -846,6 +868,11 @@ static void on_channel_connect_internal(uv_connect_t *req, int status) {
             uv_mbed_t *mbed = (uv_mbed_t *) req->handle;
             uv_mbed_read(mbed, ziti_alloc_cb, on_channel_data);
             ch->reconnect_count = 0;
+
+            //save the api_session_id that was used to request a session and create this channel
+            memset(ch->api_session_id, '\0', sizeof(ch->api_session_id));
+            strncpy(ch->api_session_id, ch->ctx->api_session->id, strlen(ch->ctx->api_session->id) - 1);
```

Comment:
`-1` is most definitely wrong

Reply:
is it leftover from debugging?

Reply:
The API session id is set during initialization and during connection. If debug output was shown while a channel before initial connect wanted it to be clear what the intended API session id was.